doc/datasets.rst

```diff
@@ -12,7 +12,8 @@ Datasets
 
 .. autosummary::
    :toctree: generated/
-
+   
+   fetch.fetch_dataset
```

Comment:
Adding the public function came out of a desire to enable downstream packages that might want to use the dataset fetching code with the same structure as MNE does. 

E.g. mne-fnirs https://github.com/mne-tools/mne-nirs/issues/386
and maybe mne-connectivity in the future.

And just any research package that might want to do CI with some private repos?

WDYT about getting rid of _data_path altogether and then just having `fetch_dataset`?, or should we keep just have `_data_path` and let downstream packages use this private function if they so desire?

Reply:
I have no opinion either way, but just noting that I use many private function in MNE-NIRS, so don't make the decision based on me.

Reply:
I leave the decisions to then @larsoner @agramfort and @drammock :p 

I'm okay either way, since now, `_data_path` is a pretty robust generalized function now.

Reply:
2 days ago we got rid of the private `_fetch_file` and it broke things downstream for mne-nirs. I think we should have a public function so that downstream packages have something less volatile to rely on.

Reply:
Exposed now `fetch_dataset` as `mne.datasets.fetch_dataset` and updated docs and init paths